routing/filter_creation_metrics.go

```diff
@@ -0,0 +1,103 @@
+package routing
+
+import (
+	"time"
+
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	maxAge              = 2
+	metricsPrefix       = "routeCreationTime."
+	defaultCreationTime = "default"
+)
+
+type FilterCreationMetrics struct {
+	metrics   filters.Metrics
+	configIds map[string]map[string]int
+}
+
+func NewFilterCreationMetrics(metrics filters.Metrics) *FilterCreationMetrics {
+	return &FilterCreationMetrics{metrics: metrics, configIds: map[string]map[string]int{}}
+}
+
+// Do implements routing.PostProcessor and records the filter creation time.
+func (m *FilterCreationMetrics) Do(routes []*Route) []*Route {
+	for _, r := range routes {
+		for name, start := range m.startTimes(r) {
+			m.metrics.MeasureSince(metricsPrefix+name, start)
+		}
+	}
+
+	m.pruneCache()
+
+	return routes
+}
+
+func (m *FilterCreationMetrics) startTimes(route *Route) map[string]time.Time {
+	startTimes := map[string]time.Time{}
+
+	t := m.configInfoStartTime(route.ConfigInfo, defaultCreationTime)
+	if !t.IsZero() {
+		startTimes[defaultCreationTime] = t
+	}
+
+	for _, f := range route.Filters {
+		t := m.filterStartTime(f)
+		o, exists := startTimes[f.Name]
```

Comment:
i don't see a reason to use the filter name as a key, because the creation time should depend only the route. This also can explode the metrics space.

Reply:
I think we should use the config ID as a key and not the filter name. This can also reduce the metrics space in case of grouped routes, like the ones coming from the same ingress object, in case of a Kubernetes scenario.

Reply:
(This also applies to the `default` key, I think it's not required.)